hazelcast/src/main/java/com/hazelcast/gcp/GcpAuthenticator.java

```diff
@@ -73,8 +73,11 @@ String refreshAccessToken(String privateKeyPath, long currentTimeMs) {
 
     private String createBody(String privateKeyPath, long currentTimeMs)
             throws Exception {
-        JsonObject privateKeyJson = Json.parse(
-                new InputStreamReader(new FileInputStream(privateKeyPath), StandardCharsets.UTF_8)).asObject();
+        JsonObject privateKeyJson;
+        try (InputStreamReader inputStreamReader = new InputStreamReader(new FileInputStream(privateKeyPath),
+                StandardCharsets.UTF_8)) {
+            privateKeyJson = Json.parse(inputStreamReader).asObject();
```

Comment:
```suggestion
        JsonObject privateKeyJson;
        try (Reader reader = Files.newBufferedReader(Paths.get(privateKeyPath,
                StandardCharsets.UTF_8))) {
            privateKeyJson = Json.parse(reader).asObject();
```

Ideally the `InputStreamReader` should be using the `Reader` interface type.

Also I prefer `Files.newBufferedReader` when the `path` is a `String` as it avoids `new InputStreamReader(new FileInputStream())` and is NIO - but that's personal preference.

Reply:
rev : c82b526236726cfeee29306f29643012d023158c